### PR TITLE
Some fixes for projectiles

### PR DESCRIPTION
**Pull request by Ikalpo**
## About The Pull Request
* fixes projectiles flying across the map when forcemove()'d, like when they hit the edge of the looping overmap
* fixes barricades causing projectiles to get stuck on them
* fixes projectiles getting stuck on the ships that fired them
## Why It's Good For The Game
fixmangud
## Changelog
:cl:
fix: projectiles edgewarp properly again
fix: projectiles no longer get stuck on barricades
fix: projectiles no longer get stuck on ships
/:cl:

### CODE REVIEW

File: nsv13/code/game/objects/structures/barricade.dm [hunk 1]
```diff
@@ -131,21 +131,7 @@
...
-					direction = EAST
-				if(NORTHWEST)
-					direction = WEST
-				if(SOUTHEAST)
-					direction = EAST
-				if(SOUTHWEST)
-					direction = WEST
-		if(!(dir & direction)) //This means they're not shooting the direction we're facing
-			return COMPONENT_ATOM_BLOCK_EXIT
+		return 0 //who cares lol
```
Comment (Karmic-Skink) [inline nsv13/code/game/objects/structures/barricade.dm:134]: Confirm that this change still retains functionality of being able to fire back over barricades.

Comment (Ikalpo) [reply nsv13/code/game/objects/structures/barricade.dm:134]: here's a bunch of screenshots, just for you
image

File: nsv13/code/modules/projectiles/projectile.dm [hunk 2]
```diff
@@ -8,9 +8,6 @@ GLOBAL_LIST_INIT(projectile_hitbox, list(new /datum/vector2d(-2,16),\
 	var/datum/component/physics2d/physics2d = null
 	var/obj/structure/overmap/overmap_firer = null
 
-/obj/item/projectile/forceMove(atom/destination)
-	return doMove(destination)
-
```
Comment (TheFakeElon) [inline nsv13/code/modules/projectiles/projectile.dm:13]: ...I'm assuming you haven't tested overmap projectiles yet

Comment (Ikalpo) [reply nsv13/code/modules/projectiles/projectile.dm:13]: I've tested them plenty, if you can tell me exactly what this is supposed to fix I can probably implement that fix without screwing up the projectile's forcemove()

Comment (TheFakeElon) [reply nsv13/code/modules/projectiles/projectile.dm:13]: Oh. Weird. If it works it works!

**Review by Bokkiewokkie — APPROVED** [review]

**Pull request merged**